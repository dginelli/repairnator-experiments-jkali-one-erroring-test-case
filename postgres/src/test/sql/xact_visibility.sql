BEGIN;
CREATE TABLE xact_test (
  id serial8 not null primary key,
  title text
);
CREATE INDEX idxxact_test ON xact_test USING zombodb(zdb('xact_test', ctid), zdb(xact_test)) WITH (url='http://localhost:9200/');

INSERT INTO xact_test (title) values ('test 1');
INSERT INTO xact_test (title) values ('test 2');
INSERT INTO xact_test (title) values ('test 3');

SELECT * FROM xact_test WHERE zdb('xact_test', ctid) ==> 'title:test*' ORDER BY id;

UPDATE xact_test SET title = 'foo' WHERE title = 'test 1';
SELECT * FROM xact_test WHERE zdb('xact_test', ctid) ==> '' ORDER BY id;

SELECT * FROM zdb_tally('xact_test', 'title', '^.*', '', 5000, 'term');
ABORT;