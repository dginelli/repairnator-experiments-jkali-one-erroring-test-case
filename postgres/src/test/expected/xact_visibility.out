BEGIN;
CREATE TABLE xact_test (
  id serial8 not null primary key,
  title text
);
CREATE INDEX idxxact_test ON xact_test USING zombodb(zdb('xact_test', ctid), zdb(xact_test)) WITH (url='http://localhost:9200/');
INSERT INTO xact_test (title) values ('test 1');
INSERT INTO xact_test (title) values ('test 2');
INSERT INTO xact_test (title) values ('test 3');
SELECT * FROM xact_test WHERE zdb('xact_test', ctid) ==> 'title:test*' ORDER BY id;
 id | title  
----+--------
  1 | test 1
  2 | test 2
  3 | test 3
(3 rows)

UPDATE xact_test SET title = 'foo' WHERE title = 'test 1';
SELECT * FROM xact_test WHERE zdb('xact_test', ctid) ==> '' ORDER BY id;
 id | title  
----+--------
  1 | foo
  2 | test 2
  3 | test 3
(3 rows)

SELECT * FROM zdb_tally('xact_test', 'title', '^.*', '', 5000, 'term');
  term  | count 
--------+-------
 foo    |     1
 test 2 |     1
 test 3 |     1
(3 rows)

ABORT;
